### DISCUSSION TITLE
How to implement multi-input (not multi-dimension input) GPR?

### DISCUSSION THREAD

**Discussion by supersjgk** [cornellius-gp/gpytorch#2435] [Q&A]
I want 5 inputs (not 5 dimensions) to give 1 output. The inputs are not independent. I cannot figure it out. Please suggest me how to do it?

**Comment by cottrell** [comment]
Can you give an example of $X$? (the input). I think I have same question but not sure. My question is about how to structure inputs of mixed types. i.e. some float features and some int (categorical) features.

**Comment by supersjgk** [comment]
Something like this 
X = torch.randn(100,5)
Y = torch.randn(100,1)

I don't think my problem(multi-inputs) aligns with yours(multi-dimensions/multi-variate). You can use multivariate distribution:  [https://docs.gpytorch.ai/en/stable/distributions.html#gpytorch.distributions.MultivariateNormal](url) and choose a combination of kernels for continuous and discrete features

**Reply by cottrell** [reply]
This is my problem. 1d output and multi-input. But I was looking for multi-input with different types. Have compromised to just cast ints to floats and store in one tensor X and cast to int just before they are used.

Your problem is the standard one. i.e. K(X, X) should just work for standard kernels and should be 100 x 100. The output dimension is determined by the target Y.shape[1] which is 1. Not sure if I'm still misunderstanding your case but it looks like the standard case.
